src/main/java/bookstore/controller/AuthController.java

```diff
@@ -0,0 +1,46 @@
+package bookstore.controller;
+
+import bookstore.dto.user.UserLoginRequestDto;
+import bookstore.dto.user.UserLoginResponseDto;
+import bookstore.dto.user.UserRegistrationRequestDto;
+import bookstore.dto.user.UserResponseDto;
+import bookstore.exception.RegistrationException;
+import bookstore.security.AuthenticationService;
+import bookstore.service.user.UserService;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@Tag(name = "Authentication", description = "Endpoints for authentication and registration")
+@RequiredArgsConstructor
+@RequestMapping(value = "/auth")
+public class AuthController {
+    private final UserService userService;
+    private final AuthenticationService authenticationService;
+
+    @PostMapping("/register")
+    @ResponseStatus(HttpStatus.CREATED)
+    @Operation(summary = "Registration",
+            description = "Add new user by email and password")
+    public UserResponseDto register(@RequestBody @Valid UserRegistrationRequestDto request)
+            throws RegistrationException {
+        return userService.register(request);
+    }
+
+    @PostMapping("/login")
+    @Operation(summary = "Authentication",
+            description = "Authenticate user by email and password")
+    @ResponseStatus(HttpStatus.OK)
```

Comment:
just FYI

`OK` is the default response status, so you can omit it

remember about this in future :)